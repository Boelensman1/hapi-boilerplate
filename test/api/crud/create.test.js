const setUpCrudTest = require('test/api/setUpCrudTest')
const { role } = require('models')

const { create } = require('api/crud')

describe('Test create crud function', () => {
  test('Insert', async () => {
    const { request, h } = await setUpCrudTest()

    request.payload = {
      name: 'Test Role',
    }
    const handler = create('role', role).handler.bind(null, request, h)
    const { body } = await handler()
    expect(body).toMatchSnapshot({
      result: {
        id: expect.any(Number),
        name: expect.any(String),
        createdAt: expect.any(Date),
      },
    })
  })
  test('Insert duplicate', async () => {
    const { request, h } = await setUpCrudTest()

    // same name already exists in db
    request.payload = {
      name: 'Test Role Dup',
    }
    const handler = create('role', role).handler.bind(null, request, h)
    await handler() // insert once
    const { body } = await handler() // insert second time
    expect(body.statusCode).toBe(409)
    expect(body).toMatchSnapshot()
  })
})
