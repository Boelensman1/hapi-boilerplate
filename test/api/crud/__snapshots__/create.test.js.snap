// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test create crud function Insert 1`] = `
Object {
  "result": Object {
    "createdAt": Any<Date>,
    "id": Any<Number>,
    "name": Any<String>,
    "post_create": 0,
    "post_delete": 0,
    "post_read": 0,
    "post_update": 0,
    "role_create": 0,
    "role_delete": 0,
    "role_read": 0,
    "role_update": 0,
    "updatedAt": null,
    "user_create": 0,
    "user_delete": 0,
    "user_read": 0,
    "user_update": 0,
  },
  "statusCode": 201,
}
`;

exports[`Test create crud function Insert duplicate 1`] = `
Object {
  "statusCode": 409,
}
`;
